test/cf/templates/tf_compute_instance_machine_type.yaml

```
@@ -0,0 +1,80 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Checks network interfaces on a compute instance against a provided whitelist of networks
+
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: tfcomputeinstancemachinetypewhitelistconstraintv1
+spec:
+  crd:
+    spec:
+      names:
+        kind: TFComputeInstanceMachineTypeWhitelistConstraintV1
+      validation:
+        openAPIV3Schema:
+          properties:
+            whitelist:
+              description: "A list of machine_types allowed, for example: ['e2-medium']"
+              type: array
+              items:
+                type: string
+  targets:
+    - target: validation.resourcechange.terraform.google.com
+      rego: |
+            #
+            # Copyright 2019 Google LLC
+            #
+            # Licensed under the Apache License, Version 2.0 (the "License");
+            # you may not use this file except in compliance with the License.
+            # You may obtain a copy of the License at
+            #
+            #      http://www.apache.org/licenses/LICENSE-2.0
+            #
+            # Unless required by applicable law or agreed to in writing, software
+            # distributed under the License is distributed on an "AS IS" BASIS,
+            # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+            # See the License for the specific language governing permissions and
+            # limitations under the License.
+            #
+
+            package templates.terraform.TFComputeInstanceMachineTypeWhitelistConstraintV1
+
+            violation[{
+            	"msg": message,
+            	"details": metadata,
+            }] {
+              resource := input.review
+              resource.type == "google_compute_instance"
+
+              action_types := {x | x := resource.change.actions[_]}
+              validated_action_types := {"create", "update"}
+              count(action_types & validated_action_types) >=1
+
+              instance := resource.change.after
+
+              whitelist := object.get(input.parameters, "whitelist", [])
```

Please use "allowlist" throughout.

```suggestion
              allowlist := object.get(input.parameters, "allowlist", [])
```